plugins/git4idea/src/git4idea/repo/GitRepositoryFiles.java

```diff
@@ -316,4 +316,9 @@ public void refresh(boolean async) {
   Collection<VirtualFile> getRootDirs() {
     return ContainerUtil.newHashSet(myMainDir, myWorktreeDir);
   }
+
+  @NotNull
+  File getBranchFile(String path) {
+    return new File(FileUtil.toSystemDependentName(myMainDir.getCanonicalPath() + File.separatorChar + path));
```

Comment:
Looks like this should be changed to 
```
  @NotNull
  File getBranchFile(@NotNull String fullBranchName) {
    return file(myMainDir.getPath() + slash(fullBranchName));
  }
```
(see other `get*File` methods and `isBranchFile()`)